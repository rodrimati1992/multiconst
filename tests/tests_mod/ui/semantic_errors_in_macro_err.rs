// ensure that .. patterns can't be used with inferred array lengths
multiconst::multiconst!{
    const [A100, ..]: [u32; _] = [3, 5, 8];
}


// ensure that .. patterns can't be used multiple times in arrays
multiconst::multiconst!{
    const [.., A100, ..]: [u32; _] = [3, 5, 8];
}


// ensure that F @ .. patterns can't be used in tuples
multiconst::multiconst!{
    const (A100, A @ ..): (u32, u32, u32, u32) = (3, 5, 8, 13);
}

// ensure that .. patterns can't be used multiple times in tuples
multiconst::multiconst!{
    const (.., A100, ..): (u32, u32, u32, u32) = (3, 5, 8, 13);
}


fn main(){}