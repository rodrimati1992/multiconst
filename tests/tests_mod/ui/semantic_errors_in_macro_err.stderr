error: cannot infer length because the pattern contains a `..`
 --> tests/tests_mod/ui/semantic_errors_in_macro_err.rs:3:23
  |
3 |     const [A100, ..]: [u32; _] = [3, 5, 8];
  |                       ^^^^^^^^
  |
  = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot use `..` multiple times in array patterns
 --> tests/tests_mod/ui/semantic_errors_in_macro_err.rs:9:22
  |
9 |     const [.., A100, ..]: [u32; _] = [3, 5, 8];
  |                      ^^
  |
  = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot use `@ ..` in tuple patterns
  --> tests/tests_mod/ui/semantic_errors_in_macro_err.rs:15:18
   |
15 |     const (A100, A @ ..): (u32, u32, u32, u32) = (3, 5, 8, 13);
   |                  ^^^^^^
   |
   = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot use `..` multiple times in tuple patterns
  --> tests/tests_mod/ui/semantic_errors_in_macro_err.rs:20:22
   |
20 |     const (.., A100, ..): (u32, u32, u32, u32) = (3, 5, 8, 13);
   |                      ^^
   |
   = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)
